orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/support/TargetServerGroup.groovy

```diff
@@ -86,9 +95,15 @@ class TargetServerGroup {
         if (!zones) {
           throw new IllegalStateException("Cannot find GCE zones in stage data ${stageData}")
         }
-        return new Location(type: Location.Type.ZONE, value: zones[0])
+        return Location.zone(zones[0])
+      }
+      if (stageData.namespace) {
```

Comment:
Now that this routine is getting more complicated, can we consolidate the logic in `locationFromServerGroup()` and `locationFromStageData()`?